packages/plugin/src/Integrations/PaymentGateways/PayPal/Controllers/BasePayPalController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Solspace\Freeform\Integrations\PaymentGateways\PayPal\Controllers;
+
+use Solspace\Freeform\Attributes\Integration\Type;
+use Solspace\Freeform\controllers\BaseApiController;
+use Solspace\Freeform\Library\Helpers\HashHelper;
+use yii\web\NotFoundHttpException;
+
+abstract class BasePayPalController extends BaseApiController
+{
+    public $enableCsrfValidation = false;
+
+    protected function getRequestItems(?string $hash = null, bool $handleFormRequest = false): array
+    {
+        if (!$hash) {
+            $hash = $this->request->getHeaders()->get('FF-PAYPAL-INTEGRATION');
+            if (!$hash) {
+                $hash = $this->request->get('integration');
+            }
+        }
+
+        if (!$hash) {
+            throw new NotFoundHttpException('Integration not found');
+        }
+
+        $ids = HashHelper::decodeMultiple($hash);
+
+        $formId = $ids[0] ?? 0;
+        $integrationId = $ids[1] ?? 0;
+        $fieldId = $ids[2] ?? 0;
+
+        $form = $this->getFormsService()->getFormById($formId);
+        if (!$form) {
+            throw new NotFoundHttpException('Form not found');
+        }
+
+        $form->disableFunctionality(['captchas']);
+        if ($handleFormRequest) {
+            $form->handleRequest($this->request, true);
+        }
+
+        $integrations = $this->getIntegrationsService()->getForForm($form, Type::TYPE_PAYMENT_GATEWAYS);
+
+        $integration = null;
+        foreach ($integrations as $int) {
+            if ($int->getId() === $integrationId) {
+                $integration = $int;
+
+                break;
+            }
+        }
+
```

Comment:
You can set the `enabled` argument to `true` to search only for enabled integrations by default.

There is a `filter` callable argument for the method, which you can use to narrow the results directly:

```suggestion
        $integration = $this->getIntegrationsService()->getFirstForForm(
            $form,
            Type::TYPE_PAYMENT_GATEWAYS,
            true,
            filter: fn ($integration) => $integration->getId() === $integrationId,
        );
```